argus/shell.py

```diff
@@ -0,0 +1,264 @@
+        if not image[0] or not image[1]:
+            images_list.remove(image)
+            print("[ERROR] Invalid image imformation id: {} - name: {}".format(
+                image[0], image[1]))
+
+    return images_list
+
+
+def _get_base_directory(directory):
+    """Return the path to the root directory for this run."""
+    if directory:
+        if not os.path.isdir(directory):
+            os.mkdir(directory)
+        directory = directory
+    else:
+        directory = tempfile.mkdtemp(prefix="argus-", dir="/tmp")
+    return os.path.abspath(directory)
+
+
+def _prepare_argument_parser():
+    """Prepare the argument parser."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-d", "--directory",
+                        help="Directory that will hold the required files.")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-i", "--image_id", action="append",
+                       help="Glance image-id to use for this run.")
+    group.add_argument("-n", "--image_name", action="append",
+                       help="Glance image-name to use for this run."
+                            "This is dangerous, be sure the name matches"
+                            " just one image.")
+    parser.add_argument("-f", "--flavor",
+                        help="The flavor id.")
+    parser.add_argument("-t", "--tests", defualt="",
+                        help="The tests you want to run.")
+    parser.add_argument("-r", "--resources", default=ci.RESOURCES_LINK,
+                        help="URL to argus resources.")
+    parser.add_argument("-p", "--parallel", type=int,
+                        help="many processes to use in parallel.")
+    parser.add_argument("-l", "--local",
+                        help="Local git repository with all the argus files.")
+    parser.add_argument("-s", "--separate", action="store_true",
+                        help="Make sepatate log files for each scenario.")
+    parser.add_argument("-g", "--git_command", default=None,
+                        help="The git command that will test the "
+                             "Cloudbase-Init patch.")
+    parser.add_argument("-z", "--zip_patch", default="",
+                        help="The zip patch to use when testing.")
+    return parser
+
+
+def _prepare_enviroment(local, directory):
+    """Prepare the temp enviroment."""
+    os.mkdir(os.path.join(directory, "ci"))
+
+    if local:
+        local = os.path.abspath(local)
+        os.link(os.path.join(local, ".testr.conf"),
+                os.path.join(directory, ".testr.conf"))
+
+        os.link(os.path.join(local, "ci", "tests.py"),
+                os.path.join(directory, "ci", "tests.py"))
+    else:
+        # Download the necessary items
+        download_argus_resource(".testr.conf", os.path.join(
+            directory, ".testr.conf"))
+
+        download_argus_resource("ci/tests.py", os.path.join(
+            directory, "ci", "tests.py"))
+
```

Comment:
How about running `python setup.py install` before `testr init`, so that we are sure we are using the latest Argus code, just a suggestion.